app/src/main/java/com/codesoom/assignment/filters/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,61 @@
+package com.codesoom.assignment.filters;
+
+import com.codesoom.assignment.application.AuthenticationService;
+import com.codesoom.assignment.errors.InvalidTokenException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class JwtAuthenticationFilter extends BasicAuthenticationFilter {
+    private final AuthenticationService authenticationService;
+
+    public JwtAuthenticationFilter(AuthenticationManager authenticationManager, AuthenticationService authenticationService) {
+        super(authenticationManager);
+        this.authenticationService = authenticationService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws IOException, ServletException {
+
+        if(filterWithPathAndMethod(request)){
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String authorization = request.getHeader("Authorization");
+
+        if (authorization == null) {
+            throw new InvalidTokenException("");
+        }
+        String accessToken = authorization.substring("Bearer ".length());
+        Long userId = authenticationService.parseToken(accessToken);
+        request.setAttribute("userId", userId);
+
+        chain.doFilter(request, response);
+    }
+
+    private boolean filterWithPathAndMethod(HttpServletRequest request) {
+        String path = request.getRequestURI();
+        if (!path.startsWith("/products")) {
+            return true;
+        }
+
+        String method = request.getMethod();
+        if (method.equals("GET")) {
```

Comment:
이런 식으로 하면 `method`가 혹시 `null`이라 하더라도 문제없죠.
```suggestion
        if ("GET".equals(method)) {
```